### ISSUE TITLE
Is' there anything wrong with levenshtein transformer when using it in a refinement task toy set(1k, train/valid/test are all the same)) like this ?

### ISSUE THREAD

**Issue by CheungZeeCn** [pytorch/fairseq#3291]
It looks quite difficult for lev-t to minimize the loss of a very simple task(design to overfit the 1k dataset, but not happend)
please help me to find out whether my steps are wrong or the model is not suitable for this task.

# task
     correction

# dataset

```
     Chinese 1k, separated by  spaces, all the dataset files(train/valid/test) are the same
     examples:
==> train.src-tgt.src <==
联 合 国 紧 急 事 务 首 席 协 调 官 艾 蒽 兰 表 示 ， 这 是 全 球 有 史 以 来 首 次 子 灾 难 发 生 候 这 么 短 一 段 时 间 内 ， 就 筹 集 到 这 么 高 的 金 额 。
日 本 大 藏 省 一 名 官 员 坚 称 ， 日 本 仍 忠 于 全 球 自 由 贸 易 贞 经 神 。

==> train.src-tgt.tgt <==
联 合 国 紧 急 事 务 首 席 协 调 官 艾 基 兰 表 示 ， 这 是 全 球 有 史 以 来 首 次 在 灾 难 发 生 后 这 么 短 一 段 时 间 内 ， 就 筹 集 到 这 么 高 的 金 额 。
日 本 大 藏 省 一 名 官 员 坚 称 ， 日 本 仍 忠 于 全 球 自 由 贸 易 的 精 神 。
```

# what have I done?
```
1. $ TEXT=datasets/news_title
2. $ fairseq-preprocess \
    --source-lang src --target-lang tgt \
    --trainpref $TEXT/train.src-tgt --validpref $TEXT/valid.src-tgt --testpref $TEXT/test.src-tgt  \
    --destdir data-bin/news_title --thresholdtgt 0 --thresholdsrc 0 \
    --workers 20
3. CUDA_VISIBLE_DEVICES=0  fairseq-train     data-bin/news_title   --save-dir news_title    --ddp-backend=no_c10d     --task translation_lev     --criterion nat_loss     --arch levenshtein_transformer_toy     --noise random_delete       --optimizer adam --adam-betas '(0.9,0.98)'     --lr 0.005 --lr-scheduler inverse_sqrt     --min-lr '1e-09' --warmup-updates 10000     --warmup-init-lr '1e-07' --label-smoothing 0.1     --dropout 0.3 --weight-decay 0.01     --decoder-learned-pos     --encoder-learned-pos     --apply-bert-init     --log-format 'simple' --log-interval 100     --fixed-validation-seed 7     --max-tokens 2000    --save-interval-updates 10000     --max-update 300000

about arch toy:
def levenshtein_transformer_toy(args):
    args.encoder_embed_dim = getattr(args, "encoder_embed_dim", 128)
    args.encoder_ffn_embed_dim = getattr(args, "encoder_ffn_embed_dim", 512)
    args.encoder_attention_heads = getattr(args, "encoder_attention_heads", 8)
    args.encoder_normalize_before = getattr(args, "encoder_normalize_before", False)
    args.decoder_embed_dim = getattr(args, "decoder_embed_dim", 128)
    args.decoder_ffn_embed_dim = getattr(args, "decoder_ffn_embed_dim", 512)
    args.decoder_attention_heads = getattr(args, "decoder_attention_heads", 8)
    args.encoder_layers = getattr(args, "encoder_layers", 4)
    args.decoder_layers = getattr(args, "decoder_layers", 4)
    args.dropout = getattr(args, "dropout", 0.1)
    args.early_exit = getattr(args, "early_exit", "2,2,2")
    levenshtein_base_architecture(args)

```

# about the loss,  cannot decline to a reasonable value (it should be overfitted and should be quite small)
```
2021-02-28 19:18:09 | INFO | fairseq.trainer | Finished saving checkpoint to news_title/checkpoint557.pt
2021-02-28 19:18:09 | INFO | fairseq.checkpoint_utils | saved checkpoint news_title/checkpoint557.pt (epoch 557 @ 14482 updates, score 3.722) (writing took 0.0769534888677299 seconds)
2021-02-28 19:18:09 | INFO | fairseq_cli.train | end of epoch 557 (average epoch stats below)
2021-02-28 19:18:09 | INFO | train | epoch 557 | loss 4.224 | nll_loss 0.801 | mask_ins 1.722 | word_ins 2.298 | word_del 0.204 | ppl 18.69 | wps 19064.4 | ups 11.3 | wpb 1687.3 | bsz 38.5 | num_updates 14482 | lr 0.00415485 | gnorm 10.204 | train_wall 2 | wall 0
2021-02-28 19:18:09 | INFO | fairseq.trainer | begin training epoch 5582021-02-28 19:18:11 | INFO | valid | epoch 558 | valid on 'valid' subset | loss 3.782 | nll_loss 0.091 | mask_ins 1.766 | word_ins 1.963 | word_del 0.053 | ppl 13.75 | wps 86815.4 | wpb 1687.3 | bsz 38.5 | num_updates 14508 | best_loss 3.569
2021-02-28 19:18:11 | INFO | fairseq_cli.train | begin save checkpoint
2021-02-28 19:18:11 | INFO | fairseq.trainer | Preparing to save checkpoint to news_title/checkpoint558.pt after 14508 updates
2021-02-28 19:18:11 | INFO | fairseq.trainer | Finished saving checkpoint to news_title/checkpoint558.pt
2021-02-28 19:18:11 | INFO | fairseq.checkpoint_utils | saved checkpoint news_title/checkpoint558.pt (epoch 558 @ 14508 updates, score 3.782) (writing took 0.14734645397402346 seconds)
2021-02-28 19:18:11 | INFO | fairseq_cli.train | end of epoch 558 (average epoch stats below)
2021-02-28 19:18:11 | INFO | train | epoch 558 | loss 3.912 | nll_loss 0.579 | mask_ins 1.645 | word_ins 2.11 | word_del 0.157 | ppl 15.05 | wps 18794.4 | ups 11.14 | wpb 1687.3 | bsz 38.5 | num_updates 14508 | lr 0.00415113 | gnorm 5.177 | train_wall 2 | wall 0
2021-02-28 19:18:11 | INFO | fairseq.trainer | begin training epoch 559
```

# output examples
```
S-306   康 提 之 从 十 二 月 一 曰 展 开 竞 选 活 动 迄 今 ， 只 参 架 过 一 次 公 开 集 会 ， 说 他 脚 痛 。
T-306   康 提 自 从 十 二 月 一 日 展 开 竞 选 活 动 迄 今 ， 只 参 加 过 一 次 公 开 集 会 ， 说 他 脚 痛 。
H-306   -0.04133198410272598    康 提 自 从 十 二 月 一 日 展 开 竞 选 活 动 迄 今 ， 只 参 加 过 一 次 公 开 集 会 ， 说 他 脚 痛 。
D-306   -0.04133198410272598    康 提 自 从 十 二 月 一 日 展 开 竞 选 活 动 迄 今 ， 只 参 加 过 一 次 公 开 集 会 ， 说 他 脚 痛 。
P-306   0.0000 -0.0036 -0.0103 -0.0340 -0.0167 -0.0704 -0.0863 -0.0183 -0.0867 -0.0608 -0.0640 -0.0362 -0.0075 -0.0178 -0.0331 -0.0446 -0.0403 -0.0382 -0.0658 -0.0432 -0.0014 -0.0900 -0.0278 -0.0900 -0.0137 -0.0281 -0.0324 -0.0034 -0.1106 -0.0705 -0.0531 -0.0705 -0.0178 -0.0021 -0.0987 0.0000
I-306   6
S-85    必 尔 斯 登 学 院 得 校 长 未 查 証 这 两 封 推 荐 函 的 真 实 性 ， 即 同 意 麦 金 农 入 校 就 读 。
T-85    毕 尔 斯 登 学 院 的 校 长 未 查 证 这 两 封 推 荐 函 的 真 实 性 ， 即 同 意 麦 金 农 入 校 就 读 。
H-85    -0.026310620829463005   毕 尔 斯 登 学 院 的 校 长 未 查 证 这 两 封 推 荐 函 的 真 实 性 ， 即 同 意 麦 金 农 入 校 就 读 。
D-85    -0.026310620829463005   毕 尔 斯 登 学 院 的 校 长 未 查 证 这 两 封 推 荐 函 的 真 实 性 ， 即 同 意 麦 金 农 入 校 就 读 。
P-85    0.0000 -0.0474 -0.0433 -0.0337 -0.0315 -0.0265 -0.0036 -0.0057 -0.0071 -0.0334 -0.0434 -0.0143 -0.0142 -0.0801 -0.0230 -0.0049 -0.0081 -0.0178 -0.0052 -0.0722 -0.0081 -0.0135 -0.0119 -0.0666 -0.0746 -0.0024 -0.0126 -0.0019 -0.0691 -0.0080 -0.0333 -0.0052 -0.0161 -0.0065 -0.1019 0.0000
I-85    2
S-95    抗 意 工 人 德 其 他 耍 求 还 包 括 提 高 薪 资 ， 并 让 一 名 遭 到 解 雇 的 工 会 领 导 人 复 职 。
T-95    抗 议 工 人 的 其 他 要 求 还 包 括 提 高 薪 资 ， 并 让 一 名 遭 到 解 雇 的 工 会 领 导 人 复 职 。
H-95    -0.07518000900745392    抗 议 工 人 德 其 他 要 求 还 包 括 提 高 薪 资 ， 并 然 一 名 遭 到 解 雇 的 工 会 领 导 人 复 职 。
D-95    -0.07518000900745392    抗 议 工 人 德 其 他 要 求 还 包 括 提 高 薪 资 ， 并 然 一 名 遭 到 解 雇 的 工 会 领 导 人 复 职 。
P-95    0.0000 -0.0130 -0.0489 -0.0712 -0.1194 -0.2567 -0.0236 -0.0475 -0.0296 -0.0328 -0.0342 -0.0243 -0.0265 -0.0147 -0.1147 -0.0146 -0.0108 -0.0633 -0.0394 -0.8606 -0.1163 -0.1918 -0.0205 -0.0263 -0.0281 -0.0097 -0.0698 -0.0356 -0.1209 -0.0029 -0.0121 -0.0988 -0.0136 -0.0024 -0.1117 0.0000
I-95    6
S-39    豪 国 再 度 抨 击 缅 甸 军 事 执 政 团 ， 指 仰 光 茌 对 抗 毒 品 生 产 和 走 私 方 面 明 显 矢 败 。
T-39    美 国 再 度 抨 击 缅 甸 军 事 执 政 团 ， 指 仰 光 在 对 抗 毒 品 生 产 和 走 私 方 面 明 显 失 败 。
H-39    -0.09020160138607025    美 国 在 度 抨 击 缅 甸 军 事 执 政 团 ， 指 仰 光 在 对 抗 毒 品 生 产 和 走 私 方 面 明 显 失 败 。
D-39    -0.09020160138607025    美 国 在 度 抨 击 缅 甸 军 事 执 政 团 ， 指 仰 光 在 对 抗 毒 品 生 产 和 走 私 方 面 明 显 失 败 。
P-39    0.0000 -0.0771 -0.1128 -0.4113 -0.0910 -0.0062 -0.0431 -0.0141 -0.1012 -0.0713 -0.1516 -0.0131 -0.0286 -0.0064 -0.0639 -0.0420 -0.0183 -0.0153 -0.1220 -0.1722 -0.0204 -0.0088 -0.0087 -0.0270 -0.0229 -0.0579 -0.0100 -1.2695 -0.0447 -0.0514 -0.0136 -0.0287 -0.0156 -0.0134 -0.0930 0.0000
I-39    2
S-782   新 加 坡 政 府 也 呼 应 全 国 工 薪 理 事 会 吁 谓 雇 主 考 卢 吧 裁 员 方 法 当 作 最 后 的 途 径 。
T-782   新 加 坡 政 府 也 呼 应 全 国 工 薪 理 事 会 吁 请 雇 主 考 虑 把 裁 员 方 法 当 作 最 后 的 途 径 。
H-782   -0.05080254003405571    新 加 坡 政 府 也 呼 应 全 国 工 薪 理 事 会 吁 请 雇 主 考 虑 把 裁 员 方 法 当 作 最 后 的 途 径 。
D-782   -0.05080254003405571    新 加 坡 政 府 也 呼 应 全 国 工 薪 理 事 会 吁 请 雇 主 考 虑 把 裁 员 方 法 当 作 最 后 的 途 径 。
P-782   0.0000 -0.0367 -0.0359 -0.0089 -0.0247 -0.0179 -0.1464 -0.0076 -0.1092 -0.0710 -0.1136 -0.0268 -0.0105 -0.0171 -0.1980 -0.1246 -0.0031 -0.0057 -0.0196 -0.0135 -0.0017 -0.0038 -0.0248 -0.0031 -0.0409 -0.0404 -0.1137 -0.0242 -0.0233 -0.0499 -0.1531 -0.0732 -0.1252 -0.0797 -0.0812 0.0000
```

datasets can be seem in attachments
 [news_title.raw.tar.gz] (https://github.com/CheungZeeCn/fairseq/blob/master/news_title.raw.tar.gz)
 [news_title.tar.gz] (https://github.com/CheungZeeCn/fairseq/blob/master/news_title.tar.gz)
